File: Museo Java/Director.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Director extends Usuario {

    public Director(int idUsuario, String nombre, String email, String contrasena) {
        super(idUsuario, nombre, email, contrasena);
    }

    public void gestionarPrestamo(Prestamo prestamo) {
        if (!getAutenticado()) {
            System.out.println("Error: Debe autenticarse para gestionar préstamos.");
            return;
        }
        prestamo.registrarPrestamo();
    }

    public double consultarValoracionTotal(Museo museo) {
        if (!getAutenticado()) {
            System.out.println("Error: Debe autenticarse para consultar valoraciones.");
            return 0.0;
        }
        double total = museo.calcularValoracionTotal();
        System.out.printf("Valoración total del museo '%s': $%,.2f%n", museo.getNombre(), total);
        return total;
    }

    public Map<String, Integer> consultarEstadisticasRestauracion(List<Restauracion> restauraciones) {
        if (!getAutenticado()) {
            System.out.println("Error: Debe autenticarse para consultar estadísticas.");
            return new HashMap<>();
        }
        int enCurso = 0, finalizadas = 0;
        for (Restauracion r : restauraciones) {
            if ("en_curso".equals(r.getEstado())) enCurso++;
            else if ("finalizada".equals(r.getEstado())) finalizadas++;
        }
        Map<String, Integer> stats = new HashMap<>();
        stats.put("total", restauraciones.size());
        stats.put("en_curso", enCurso);
        stats.put("finalizadas", finalizadas);
        return stats;
    }
}

File: MuseoPOO/director.py
"""Módulo que define la clase Director."""

from __future__ import annotations

from typing import TYPE_CHECKING

from usuario import Usuario

if TYPE_CHECKING:
    from museo import Museo
    from prestamo import Prestamo


class Director(Usuario):
    """Representa al director del museo."""

    def __init__(
        self,
        id_usuario: int,
        nombre: str,
        email: str,
        contrasena: str,
    ) -> None:
        super().__init__(id_usuario, nombre, email, contrasena)

    def gestionar_prestamo(self, prestamo: Prestamo) -> None:
        """Gestiona un préstamo de obra de arte."""
        if not self.get_autenticado():
            print(
                "Error: Debe autenticarse para gestionar "
                "préstamos."
            )
            return
        prestamo.registrar_prestamo()

    def consultar_valoracion_total(self, museo: Museo) -> float:
        """Consulta la valoración total de las obras del museo."""
        if not self.get_autenticado():
            print(
                "Error: Debe autenticarse para consultar "
                "valoraciones."
            )
            return 0.0
        total = museo.calcular_valoracion_total()
        print(
            f"Valoración total del museo "
            f"'{museo.get_nombre()}': ${total:,.2f}"
        )
        return total

    def consultar_estadisticas_restauracion(
        self, restauraciones: list,
    ) -> dict:
        """Consulta estadísticas de restauraciones."""
        if not self.get_autenticado():
            print(
                "Error: Debe autenticarse para consultar "
                "estadísticas."
            )
            return {}
        en_curso = sum(
            1 for r in restauraciones
            if r.get_estado() == "en_curso"
        )
        finalizadas = sum(
            1 for r in restauraciones
            if r.get_estado() == "finalizada"
        )
        return {
            "total": len(restauraciones),
            "en_curso": en_curso,
            "finalizadas": finalizadas,
        }
